pkg/engine/context/imageutils.go

```diff
@@ -43,124 +42,96 @@ func (i *ImageInfo) String() string {
 }
 
 type ContainerImage struct {
-	Name  string
-	Image *ImageInfo
+	ImageInfo
+	Name string
 }
 
 type Images struct {
-	InitContainers      map[string]*ImageInfo `json:"initContainers,omitempty"`
-	Containers          map[string]*ImageInfo `json:"containers"`
-	EphemeralContainers map[string]*ImageInfo `json:"ephemeralContainers"`
+	InitContainers      map[string]ImageInfo `json:"initContainers,omitempty"`
```

Comment:
Why not use the reference here?

Reply:
The map is a reference but why entries should be nullable ?
I mean all map entries should be valid.